api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+        }
+        return validateResponse;
+    }
+
+    @Override
+    public AccountValidationResponse deleteAccount(String accountId) {
+        LOGGER.info(DELETING_ACCOUNT);
+        AccountValidationResponse response;
+        try {
+            BasicSessionCredentials credentials = credentialProvider.getBaseAccCredentials();
+            Regions region = Regions.fromName(System.getenv(REGION));
+            String roleName = System.getenv(PALADINCLOUD_RO);
+            AWSSecretsManager secretClient = AWSSecretsManagerClientBuilder
+                    .standard()
+                    .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                    .withRegion(region).build();
+            DeleteSecretRequest deleteRequest = new DeleteSecretRequest().withSecretId(secretManagerPrefix +
+                    FORWARD_SLASH + roleName + FORWARD_SLASH + serviceType() +
+                    FORWARD_SLASH + accountId).withForceDeleteWithoutRecovery(true);
+            DeleteSecretResult deleteResponse = secretClient.deleteSecret(deleteRequest);
+            LOGGER.info(DELETE_SECRET_RESPONSE, deleteResponse);
+            //delete entry from db
+            response = deleteAccountFromDB(accountId);
+            response.setType(Constants.REDHAT);
+            List<AccountDetails> onlineAccounts = findOnlineAccounts(STATUS_CONFIGURED, Constants.REDHAT);
+            if (onlineAccounts == null || onlineAccounts.isEmpty()) {
+                LOGGER.debug(ACCOUNTS_DELETED_MSG);
+                updateConfigProperty(REDHAT_ENABLED, FALSE, JOB_SCHEDULER);
+            }
+        } catch (Exception e) {
+            response = new AccountValidationResponse();
+            response.setValidationStatus(FAILURE);
+            response.setMessage(String.format(ERROR_MESSAGE_TEMPLATE, DELETING));
+            response.setErrorDetails(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), DELETING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE, DELETING));
+            LOGGER.info(String.format(ERROR_MESSAGE_TEMPLATE + e.getMessage(), DELETING));
+        }
+        return response;
+    }
+
+    private AccountValidationResponse validateRequest(CreateAccountRequest accountData) {
+        AccountValidationResponse response = new AccountValidationResponse();
+        StringBuilder validationErrorDetails = new StringBuilder();
+        if (accountData.getRedhatAccountId() == null || accountData.getRedhatAccountId().isEmpty()) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, PARAM_ACCOUNT_ID));
+        }
+
+        if (accountData.getRedhatAccountName() != null && accountData.getRedhatAccountName().trim().isEmpty()) {
+            validationErrorDetails.append(String.format(MISSING_MANDATORY_PARAMETER, PARAM_ACCOUNT_NAME));
+        }
+
+        if (accountData.getRedhatToken() == null || accountData.getRedhatToken().isEmpty()) {
```

Comment:
what if token is " "? will isEmpty() return true?

All of the validations are on a String value, please write a common function that tell you if given String is empty.

Also, you can simply use isBlank() as of Java 11 : https://howtodoinjava.com/java11/check-blank-string/#:~:text=isBlank()%20method%20to%20determine,isEmpty()%20method.

Reply:
if token is empty then, we will catch it at the time of URL validation
yeah but Im still using using Java 8